EXILED/Exiled.Events/EventArgs/Player/UsingMicroHIDEnergyEventArgs.cs

```diff
@@ -17,32 +17,24 @@ namespace Exiled.Events.EventArgs.Player
     /// <summary>
     /// Contains all information before MicroHID energy is changed.
     /// </summary>
-    public class UsingMicroHIDEnergyEventArgs : IPlayerEvent, IDeniableEvent, IItemEvent
+    public class UsingMicroHIDEnergyEventArgs : IDeniableEvent, IItemEvent
     {
         /// <summary>
         /// Initializes a new instance of the <see cref="UsingMicroHIDEnergyEventArgs" /> class.
         /// </summary>
-        /// <param name="player">
-        /// <inheritdoc cref="Player" />
-        /// </param>
         /// <param name="microHIDitem">
         /// <inheritdoc cref="MicroHID" />
         /// </param>
-        /// <param name="currentState">
-        /// <inheritdoc cref="CurrentState" />
-        /// </param>
         /// <param name="drain">
         /// <inheritdoc cref="Drain" />
         /// </param>
         /// <param name="isAllowed">
         /// <inheritdoc cref="IsAllowed" />
         /// </param>
-        public UsingMicroHIDEnergyEventArgs(Player player, MicroHIDItem microHIDitem, HidState currentState, float drain, bool isAllowed = true)
+        public UsingMicroHIDEnergyEventArgs(MicroHIDItem microHIDitem, float drain, bool isAllowed = true)
         {
-            Player = player;
             MicroHID = Item.Get<MicroHid>(microHIDitem);
-            CurrentState = currentState;
-            Drain = drain;
+            Drain = MicroHID.Energy - drain;
```

Comment:
are you sure of this it's fell like you made it wrong

Reply:
Drain is a amount that will be removed. In method amount of new energy is saved. So yeah that should work